02-peer-protocol.md

```diff
@@ -263,14 +263,20 @@ The sender MUST wait until the funding transaction has reached
 `minimum-depth` before sending this message.  The sender MUST encode
 the block position of the funding transaction into `channel-id`.  If
 the sender has already received `funding_locked` from the other node,
-it MUST fail the channel if its own `channel-id` does not match the
-received.  The sender MUST set `next-per-commitment-point` to the
+it MAY fail the channel if its own `channel-id` does not match the
+received: otherwise it MUST ignore the `funding_locked` message.
+The sender MAY re-transmit `funding_locked` if the `channel-id` changes.
```

Comment:
I think we also need to specify that a node MUST retransmit the `funding_locked` message if the `channel-id` changes, otherwise we may never make progress. If both nodes silently drop the `funding_locked` message, but never issue send the changed `channel-id` then we just sit there and wait forever. This may also include sending a `funding_locked` at a bigger depth than what we were waiting for.